### PR TITLE
Compare using #oldDefinition both inst- and class-side.

### LINKED ISSUE #5124: Epicea: Bug in EpHasImpactFilter on class changes
Steps to reproduce:
1. create a class
2. open Code Changes browser and click on the current log
3. choose "Hide changes if apply operations is redundant".
4. the class addition must be hidden (but it's not)

### PR DESCRIPTION
**Pull request by tinchodias**
Fix #5124

### CODE REVIEW

File: src/EpiceaBrowsers-Tests/EpFilterTest.class.st [hunk 1]
```diff
@@ -15,27 +15,11 @@ EpFilterTest >> classAddition [
...
-EpFilterTest >> diverseEvents [
-	^ {
-		self classAddition.
-		self expressionEvaluation.
-		self methodAddition.
-		self redo.
-		self refactoring.
-		}
-]
-
```
Comment (tinchodias) [inline src/EpiceaBrowsers-Tests/EpFilterTest.class.st:28]: This was an unused method

File: src/EpiceaBrowsers-Tests/EpFilterTest.class.st [hunk 2]
```diff
@@ -15,27 +15,11 @@ EpFilterTest >> classAddition [
...
 { #category : #running }
 EpFilterTest >> entryWith: anEvent [
 	^ OmEntry content: anEvent
 ]
 
-{ #category : #resources }
-EpFilterTest >> expressionEvaluation [
-	^ self entryWith: (EpExpressionEvaluation expression: '42')
-]
-
```
Comment (tinchodias) [inline src/EpiceaBrowsers-Tests/EpFilterTest.class.st:38]: This one too

**Comment by tinchodias** [comment]
@MarcusDenker @Ducasse this PR was closed by mistake AFAIU

**Comment by tinchodias** [comment]
(I don't have permissions to re-open it)

**Comment by estebanlm** [comment]
recheck

**Comment by MarcusDenker** [comment]
I changed it to be for Pharo9

File: src/EpiceaBrowsers/EpHasImpactVisitor.class.st [hunk 3]
```diff
@@ -48,8 +48,9 @@ EpHasImpactVisitor >> visitBehaviorCategoryChange: aChange [
 
 { #category : #visitor }
 EpHasImpactVisitor >> visitBehaviorChange: aChange [
-	self behaviorNamed: aChange behaviorAffectedName 
-		ifPresent: [ :behavior | ^ behavior definition ~= aChange behaviorAffected definitionSource ].
+	self behaviorNamed: aChange behaviorAffectedName ifPresent: [ :behavior | 
+		^ behavior oldDefinition ~= aChange behaviorAffected definitionSource or: [
+			behavior classSide oldDefinition ~= aChange behaviorAffected classSide definitionSource ] ].
```
Comment (tinchodias) [inline src/EpiceaBrowsers/EpHasImpactVisitor.class.st:53]: Here is the fix

**Comment by tinchodias** [comment]
I think this is still fine for merge:
- fixes the issue (with 1 line added in EpHasImpactVisitor>>visitBehaviorCategoryChange:)
- adds related tests 
- removes 2 old unsent methods (in tests)

File: src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	#name : #EpHasImpactFilterTest,
+	#superclass : #TestCase,
+	#instVars : [
+		'classFactory',
+		'impactFilter'
+	],
+	#category : #'EpiceaBrowsers-Tests-Filters'
+}
+
+{ #category : #tests }
```
Comment (guillep) [inline src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st]: This should be "running"

File: src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+EpHasImpactFilterTest >> setUp [
+	classFactory := ClassFactoryForTestCase new.
+	impactFilter := EpImpactCodeChangeFilter new
+		environment: self class environment;
+		yourself.
+
+]
+
+{ #category : #tests }
```
Comment (guillep) [inline src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st]: This should be "running"

File: src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	#instVars : [
+		'classFactory',
+		'impactFilter'
+	],
+	#category : #'EpiceaBrowsers-Tests-Filters'
+}
+
+{ #category : #tests }
+EpHasImpactFilterTest >> setUp [
+	classFactory := ClassFactoryForTestCase new.
```
Comment (guillep) [inline src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st:14]: Setup should call super setup.

File: src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	impactFilter := EpImpactCodeChangeFilter new
+		environment: self class environment;
+		yourself.
+
+]
+
+{ #category : #tests }
+EpHasImpactFilterTest >> tearDown [
+	classFactory cleanUp
```
Comment (guillep) [inline src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st]: TearDown should call tearDown at the end

**Review by guillep — CHANGES_REQUESTED** [review]
Hi @tinchodias thanks for the change.

I did take a look at the tests failures, and they are broken due to bad categorization of tests + two other silly things.

image

Would you mind fixing my comments? (pro-tip: you can edit the files in github if you're brave enough).

If you do, I'll integrate right away.

Thanks again!

File: src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st (hunk 7 above)
Comment (tinchodias) [reply src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st]: done

File: src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st (hunk 5 above)
Comment (tinchodias) [reply src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st]: done

File: src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st (hunk 4 above)
Comment (tinchodias) [reply src/EpiceaBrowsers-Tests/EpHasImpactFilterTest.class.st]: done

**Review by guillep — APPROVED** [review]
Thanks @tinchodias , this is great :)

tests are green

**Pull request merged**